spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2000,6 +2000,16 @@ def hide_tooltip(self):
         self.tooltip_widget.hide()
         self.clear_extra_selections('code_analysis_highlight')
 
+    def get_style_warning_sources(self):

```

Comment:
Could this possibly be moved to a separate file/static method? (Maybe that is what you are referring to in the comments.) It would be useful to be able to obtain this for #9836. In fact, one may even include
https://github.com/spyder-ide/spyder/blob/c188748789c8b069b5fc383061219f875fe36959/spyder/plugins/editor/widgets/codeeditor.py#L1971-L1977
there (see end of #9924), and possibly a convenience function for getting the correct icon providing DiagnosticSeverity and source (not sure if it is enough to instantiate them once when it comes to style changes etc)?

Reply:
Also, the menu can use the same function, which I do not think it does at the moment.

Reply:
Yes @oscargus there will be a single place to store this info, not related to this widget. But that requires finishing another PR, which will not happen for at least 3 weeks. 

I need to work on something else and then I will tackle that and come back to this ;-).

Reply:
>(not sure if it is enough to instantiate them once when it comes to style changes etc)

#9782 will allow to have style changes without a restart, so some methods will need to be called again if style changes. Since the theme can have a "dirty" flag if style change is requested, then we can have a cache of the style that is refreshed on style change to avoid recreating objects like icons, although for what I have seen Icons are relatively lightweight.